argus/introspection/cloud/windows.py

```diff
@@ -240,3 +243,28 @@ def get_cloudconfig_executed_plugins(self):
             content = self.get_instance_file_content(path)
             files[basefile] = content.strip()
         return files
+
+    def get_nics(self):
+        """Get IP available instance network adapters."""
+        # Retrieve utility script.
+        cmd = ("powershell Invoke-WebRequest -uri "
+               "{}/windows/network_details.ps1 -outfile "
+               "C:\\network_details.ps1".format(CONF.argus.resources))
+        self.remote_client.run_command_with_retry(cmd)
+        # Run and parse the output, where each adapter details
+        # block is separated by a new line.
+        cmd = "powershell C:\\network_details.ps1"
+        output = self.remote_client.run_command_verbose(cmd).strip()
```

Comment:
I have to select what I need and print them in the order I choose. Also, there are other formatting and parsing purposes based on the simplicity of the values.